File: py_cpp_data_science_workflow/py/ngram_lang_detect_run.py
import pickle
import sys
from NGramLangDetect import NGramLangDetectModel

if len(sys.argv) != 2:
  print("Usage:", sys.argv[0], "model.pkl")
  print(" - once loaded, the program reads from stdin")
  print(" - when a newline is encountered, the predicted language is " +
      "printed")
  print(" - ctrl+c or ctrl+d to exit")
  sys.exit(1)

model_file = sys.argv[1]
model = pickle.load(open(model_file, 'rb'))

while True:
  try:
    line = input()
    class_prediction = model.predict(line)
    print(model.class_name(class_prediction[0]))
  except EOFError:
    break

File: py_cpp_data_science_workflow/cpp/ngram_langdetect_run.cpp
#include <mlpack.hpp>
#include "ngram_langdetect_model.hpp"

using namespace std;
using namespace arma;
using namespace mlpack;

int main(int argc, char** argv)
{
  if (argc != 2)
  {
    cerr << "Usage: " << argv[0] << " model_file.bin" << endl;
    cerr << " - once loaded, the program reads from stdin" << endl;
    cerr << " - when a newline is encountered, the predicted language is "
        << "printed" << endl;
    cerr << " - ctrl+c or ctrl+d to exit" << endl;
    return 1;
  }

  const std::string modelFile = argv[1];

  NGramLangDetectModel m;
  data::Load(modelFile, "model", m, true /* fatal */);

  while(true)
  {
    string line;
    getline(cin, line);

    if (cin.eof())
    {
      return 0;
    }

    size_t classPrediction = m.Classify(line);
    cout << m.ClassName(classPrediction) << endl;
  }
}
